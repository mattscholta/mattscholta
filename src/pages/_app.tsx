import * as React from 'react';
import { CookiesProvider } from 'react-cookie';
import type { AppContext, AppProps } from 'next/app';
import App from 'next/app';
import parser from 'ua-parser-js';
import Cookies from 'universal-cookie';

import { intro } from '@src/config/intro';
import { IS_BROWSER } from '@src/config/settings';
import { AppComponent } from '@src/services/app/components/AppComponent';

// import { checkBasicAuth } from '@src/utils/auth';
import '@src/styles/index.css';

interface ApplicationProps extends AppProps {
  cookies: Record<string, any>;
  global: any;
}

/**
 * @name Application
 * @external https://nextjs.org/docs/advanced-features/custom-app
 * @description Next.js uses the App component to initialize pages.
 * You can override it and control the page initialization.
 */
const Application: React.FC<ApplicationProps> = (props, _context) => {
  const { cookies, global, Component, pageProps } = props;

  // Setup
  const cookieData = new Cookies(cookies);
  const cookieProps = IS_BROWSER ? {} : { cookies: cookieData };

  // Life Cycle
  React.useEffect(() => {
    intro();
  }, []);

  return (
    <CookiesProvider {...cookieProps}>
      <AppComponent global={global} pageProps={pageProps}>
        <Component {...pageProps} />
      </AppComponent>
    </CookiesProvider>
  );
};

/**
 * Only uncomment this method if you have blocking data requirements for
 * every single page in your application. This disables the ability to
 * perform automatic static optimization, causing every page in your app to
 * be server-side rendered.
 *
 * We connect to our GraphQL instance, potentially, on any page. This means
 * that any of our `useQuery` or `useMutation` actions will execute on the
 * server side, so we have to make our cookies available to the server
 */
(Application as any).getInitialProps = async (appContext: AppContext) => {
  const { ctx } = appContext;
  const { req, res: _res } = ctx;

  // Setup
  const cookies = IS_BROWSER ? document.cookie : (req as any)?.cookies ?? {};
  const global = {};
  const userAgent = req?.headers['user-agent'] ?? '';
  const ua = parser(userAgent);

  // Check auth will call res.end if password not matching.
  // if (req && res) checkBasicAuth(req, res);

  // Calls page's `getInitialProps` and fills `appProps.pageProps`
  const appProps = await App.getInitialProps(appContext);

  return { ...appProps, cookies, global, ua };
};

export { Application, Application as default };
export type { ApplicationProps };
