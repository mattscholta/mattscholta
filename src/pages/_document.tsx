import * as React from 'react';
import Document, {
  DocumentProps,
  Head,
  Html,
  Main,
  NextScript
} from 'next/document';

import { GOOGLE_TAG_MANAGER } from '@src/config/settings';
import { GoogleTagManagerIFrame } from '@src/services/tracking/components/GoogleTagManagerIFrame';
import { script as scriptArtwork } from '@src/utils/artwork';

/**
 * @name AppDocument
 * @external https://nextjs.org/docs/advanced-features/custom-document
 * @description A custom Document is commonly used to augment your
 * application's <html> and <body> tags.
 */
class AppDocument extends Document {
  render(): JSX.Element {
    return (
      <Html lang="en">
        <Head />
        <body>
          <div dangerouslySetInnerHTML={{ __html: scriptArtwork }} />
          <GoogleTagManagerIFrame id={GOOGLE_TAG_MANAGER} />
          <Main />
          <NextScript />
        </body>
      </Html>
    );
  }
}

export { AppDocument, AppDocument as default };
export type { DocumentProps as AppDocumentProps };
