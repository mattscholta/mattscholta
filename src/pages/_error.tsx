import * as React from 'react';
import classnames from 'classnames';
import Head from 'next/head';

import { SITE_TITLE } from '@src/config/constants';

import styles from './_error.module.css';

interface ErrorPageProps {}

/**
 * @name ErrorPage
 * @description Simple 404 page
 */
const ErrorPage: React.FC<ErrorPageProps> = (_props) => {
  // Styles
  const cssComponent = classnames('ui-main', styles.component);

  // Handlers
  const onClickRefresh = (event: React.MouseEvent<HTMLAnchorElement>) => {
    event.preventDefault();
    window.location.reload();
  };

  return (
    <>
      <Head>
        <title>Page not found | {SITE_TITLE}</title>
        <link rel="icon" href="/favicon.ico" />
      </Head>

      <main className={cssComponent}>
        <h1 className="ui-heading u-mb-2x">¯\_(ツ)_/¯</h1>
        <p>Sorry, we couldn’t find the page you were looking for.</p>
        <div className="u-flex">
          <a className="ui-link" href="/">
            Homepage
          </a>
          <span className="u-mx-1x">|</span>
          <a className="ui-link" href="/" onClick={onClickRefresh}>
            Try again
          </a>
        </div>
      </main>
    </>
  );
};

export { ErrorPage, ErrorPage as default };
export type { ErrorPageProps };
