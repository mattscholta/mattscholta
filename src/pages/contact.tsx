import * as React from 'react';
import classnames from 'classnames';
import Head from 'next/head';

import {
  SITE_EMAIL_ADDRESS,
  SITE_EMAIL_LINK,
  SITE_TITLE
} from '@src/config/constants';

import styles from './contact.module.css';

interface ContactProps {}

/**
 * @name Contact
 * @description Application landing page (homepage)
 */
const Contact: React.FC<ContactProps> = (_props) => {
  // Styles
  const cssComponent = classnames('ui-main', styles.component);

  // Markup
  const renderedEmail = (
    <a className="ui-link" href={SITE_EMAIL_LINK}>
      {SITE_EMAIL_ADDRESS}
    </a>
  );

  return (
    <>
      <Head>
        <title>Contact | {SITE_TITLE}</title>
      </Head>

      <main className={cssComponent}>
        <div className="ui-container-xl u-p-2x">
          <h1 className={styles.h1}>Contact Info</h1>
          <p>
            Lorem ipsum dolor sit amet consectetur adipisicing elit. Quis saepe
            ut magni id deserunt labore dignissimos at quas asperiores ipsum ab
            quos accusamus praesentium atque magnam recusandae distinctio,
            tempore voluptatum?
          </p>
          <p>Contact Jax at {renderedEmail}...</p>
        </div>
      </main>
    </>
  );
};

export { Contact as default, Contact };
export type { ContactProps };
