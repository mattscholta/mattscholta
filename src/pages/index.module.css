.component {
  margin-top: var(--grid-5x);
}

.component aside {
  flex: 0 0 400px;
}

.component section {
  flex: 1 0 320px;
}

.component p em {
  font-weight: bold;
}

.button {
  margin: var(--grid-10x) auto var(--grid-3x);
}

.hero {
  margin-bottom: var(--grid-4x);
}

.left {
  margin-bottom: theme('spacing.8x');
}

.link {
  display: block;
  font-weight: var(--font-weight-bold);
  margin-left: var(--grid-2x);
  margin-top: var(--grid-4x);
  text-align: center;
  transition: padding 200ms ease-out;
}

.link:hover {
  padding-left: var(--grid-1x);
}

.sidebar {
  margin-bottom: var(--grid-6x);
}

@media (max-width: 1024px) {
  .component {
    margin-top: 0;
  }
}

@media (max-width: 768px) {
  .left {
    margin-bottom: theme('spacing.4x');
  }

  .link {
    margin-bottom: var(--grid-6x);
  }

  .sidebar {
    margin-bottom: var(--grid-4x);
    margin-top: var(--grid-2x);
  }
}
