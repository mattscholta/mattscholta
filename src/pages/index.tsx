import * as React from 'react';
import classnames from 'classnames';
import Head from 'next/head';
import Link from 'next/link';

import { ConnectOnline } from '@src/components/ConnectOnline';
import { DevelopmentStack } from '@src/components/DevelopmentStack';
import { WorkHistory } from '@src/components/WorkHistory';
import { CLIENT_CLOUDINARY, SITE_TITLE } from '@src/config/constants';
import { getSrcSet } from '@src/utils/links';

import styles from './index.module.css';

interface IndexProps {
  posts: any[];
}

/**
 * @name Index
 * @description Application landing page (homepage)
 */
const Index: React.FC<IndexProps> = (props) => {
  const { posts: _posts } = props;

  // Setup
  const img = `${CLIENT_CLOUDINARY}/v1616195587/mattscholta/general/coast_1500.webp`;
  const sources = getSrcSet(img);

  // Styles
  const cssComponent = classnames('ui-main', styles.component);

  return (
    <>
      <Head>
        <title>About | {SITE_TITLE}</title>
      </Head>

      <main className={cssComponent}>
        <div className="ui-container-xl u-p-2x">
          <div className="ui-reflow">
            <div className={styles.left}>
              <blockquote style={{ maxWidth: '85%' }}>
                A <em>Software Engineer</em> passionate about{' '}
                <b>quality code</b> written for <b>humans</b>, unlocking{' '}
                <b>developer productivity</b>, and creating delightful{' '}
                <b>user experiences</b>.<cite>- Matthew Scholta</cite>
              </blockquote>
              <img
                alt="The California Coast"
                className={styles.hero}
                height={500}
                loading="eager"
                sizes="(min-width: 768px) 60vw, 100vw"
                src={img}
                srcSet={sources}
                title="Life's a beach, work hard and play in the sand!"
                width={1500}
              />

              <h1 className="ui-heading">Opportunity awaits üßë‚Äçüíª</h1>
              <sub className="ui-subhead" />

              <p>
                After <b>5+ great</b> years pushing the envelope at{' '}
                <a className="u-font-bold" href="https://thredup.com">
                  thredUP.com
                </a>{' '}
                the time has come for a new set of challenges. With a focus on
                front end development, user-experience, user-interaction and a
                passion for developer productivity I am beyond excited to see
                what opportunities the future holds.
              </p>

              <p>
                The technology and tools listed on this page are my preferred
                stack as of 2022. This year I'll be launching a new version of a
                long time project{' '}
                <a className="u-font-bold" href="https://barguide.io">
                  BarGuide.io
                </a>{' '}
                with all that <b>"free time"</b> I have. ü§£
              </p>

              <p>
                So, if you're interested in learning more, ask the nearest
                developer to help you find my <b>email address</b>. I've hidden
                it into the website, twice actually... Otherwise reach out and
                connect on LinkedIn.
              </p>

              <Link href="/portfolio">
                <a className={styles.link}>View Portfolio &#8594;</a>
              </Link>
            </div>

            <div className="ui-spacer" />

            <aside className={styles.right}>
              <ConnectOnline />
              <WorkHistory />
              <DevelopmentStack />
            </aside>
          </div>
        </div>
      </main>
    </>
  );
};

export { Index as default, Index };
export type { IndexProps };
