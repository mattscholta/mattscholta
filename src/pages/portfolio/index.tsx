import * as React from 'react';
import classnames from 'classnames';
import { gql } from 'graphql-request';
import { GetServerSideProps } from 'next';
import Head from 'next/head';

import { PreviewCard } from '@src/components/PreviewCard';
import { SITE_TITLE } from '@src/config/constants';
import { data as portfolio } from '@src/data/portfolio';
import { graphcms } from '@src/utils/graphcms';

import styles from './index.module.css';

interface IndexProps {
  data: any[];
}

/**
 * @name Index
 * @description Application landing page (homepage)
 */
const Index: React.FC<IndexProps> = (props) => {
  const { data } = props;
  console.log(` ðŸ’¬ ~ data`, data);

  // Styles
  const cssComponent = classnames('ui-container-xl ui-main', styles.component);
  const cssCards = classnames('u-flex u-flex-wrap', styles.cards);

  // Markup
  const renderPreview = (node: any) => {
    const { overview, slug, thumbnailTemp, title } = node;

    return (
      <PreviewCard
        description={overview}
        image={thumbnailTemp}
        key={slug}
        slug={slug}
        title={title}
      />
    );
  };

  return (
    <>
      <Head>
        <title>Index | {SITE_TITLE}</title>
      </Head>

      <div className={cssComponent}>
        <h1 className="u-visually-hidden">ListV2</h1>
        <div className={cssCards}>{data.map(renderPreview)}</div>
      </div>
    </>
  );
};

export const getServerSideProps: GetServerSideProps = async (_context) => {
  const QUERY = gql`
    {
      portfolios {
        overview
        slug
        title
        thumbnailTemp
      }
    }
  `;

  const query = await graphcms.request(QUERY);
  const { portfolios } = query;

  return {
    props: { data: portfolios }
  };
};

export { Index as default, Index };
export type { IndexProps };
