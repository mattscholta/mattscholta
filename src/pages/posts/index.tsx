import * as React from 'react';
import classnames from 'classnames';
import { gql } from 'graphql-request';
import { GetServerSideProps } from 'next';
import Head from 'next/head';

import { BlogPreviewColumn } from '@src/components/BlogPreviewColumn';
import { SITE_TITLE } from '@src/config/constants';
import { graphcms } from '@src/utils/graphcms';

import styles from './index.module.css';

interface IndexProps {
  posts: any[];
}

/**
 * @name Index
 * @description Application landing page (homepage)
 */
const Index: React.FC<IndexProps> = (props) => {
  const { posts } = props;

  // Styles
  const cssComponent = classnames('ui-main', styles.component);
  const cssHeading = classnames('u-visually-hidden', styles.heading);
  const cssPreviews = classnames('u-flex u-flex-wrap', styles.previews);

  // Markup
  const renderPreview = (post: any) => {
    const { slug } = post;

    return (
      <BlogPreviewColumn
        className={styles.preview}
        key={slug}
        post={post}
        small={false}
      />
    );
  };

  return (
    <>
      <Head>
        <title>Index | {SITE_TITLE}</title>
      </Head>

      <main className={cssComponent}>
        <div className="ui-container-xl">
          <h1 className={cssHeading}>A Developers Ramblings</h1>
          <br />
          <blockquote className="u-px-2x">
            Always leave the <span className="u-line-through">campground</span>{' '}
            <b>code</b> cleaner than you found it. üèïÔ∏è
            <cite>A Developers Ramblings</cite>
          </blockquote>

          <aside className={cssPreviews}>
            {posts.map(renderPreview)}
            {/* <BlogDisclaimer /> */}
            {/* {related && related.map(renderRelatedPosts)} */}
            {/* <BlogSharing /> */}
          </aside>
        </div>
      </main>
    </>
  );
};

export const getServerSideProps: GetServerSideProps = async (_context) => {
  const QUERY = gql`
    {
      posts {
        id
        imageTemp
        sticky
        slug
        title
      }
    }
  `;

  const query = await graphcms.request(QUERY);
  const { posts } = query;

  return {
    props: { posts }
  };
};

export { Index as default, Index };
export type { IndexProps };
