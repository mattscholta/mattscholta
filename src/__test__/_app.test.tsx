import * as React from 'react';
import { render, RenderResult } from '@testing-library/react';
import { RecoilRoot } from 'recoil';

import { Application, ApplicationProps } from '../pages/_app';

jest.mock('next/router', () => ({
  useRouter() {
    return {
      asPath: '',
      pathname: '/',
      query: {},
      route: '/'
    };
  }
}));

describe.skip('Application', () => {
  const gtag = jest.fn();

  let props: ApplicationProps;
  let wrapper: RenderResult;

  const MockComponent = () => <div>mock component</div>;
  const createWrapper = () => (
    <RecoilRoot>
      <Application {...props} />
    </RecoilRoot>
  );

  beforeAll(() => {
    window.gtag = gtag;
    props = {
      Component: MockComponent,
      cookies: {},
      global: {},
      pageProps: {},
      router: {
        asPath: '',
        pathname: '/',
        query: {},
        route: '/'
      } as any
    };

    wrapper = render(createWrapper());
  });

  it('matches snapshot', () => {
    expect(wrapper.container.childNodes).toMatchSnapshot();
  });
});
