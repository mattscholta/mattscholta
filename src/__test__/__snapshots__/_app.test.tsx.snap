// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Application matches snapshot 1`] = `
NodeList [
  <div
    class="component"
  >
    <div
      class="ui-container-xl u-flex"
    >
      <button
        class="menu"
        type="button"
      >
        <svg
          height="20"
          id="my_element"
          viewBox="0 0 20 20"
          width="20"
        >
          <title>
            Menu
          </title>
          <rect
            fill="#fff"
            height="2.5"
            width="20"
            x="0%"
            y="0%"
          />
          <rect
            fill="#fff"
            height="2.5"
            width="20"
            x="0%"
            y="8.75"
          />
          <rect
            fill="#fff"
            height="2.5"
            width="20"
            x="0%"
            y="17.5"
          />
        </svg>
      </button>
      <a
        class="u-mr-2x"
        href="/"
      >
        <svg
          class="svgBrand"
          height="59"
          viewBox="0 0 100 59"
          width="100"
        >
          <title>
            mattscholta
          </title>
          <g
            fill="none"
            fill-rule="evenodd"
            stroke="none"
            stroke-width="1"
          >
            <path
              d="M72.573561,53.8100014 L67.7974434,43 L63.0213258,53.8100014 L57.7974434,53.8100014 L52.573561,53.8100014 L47.7974434,43 L43.0213258,53.8100014 L24.0228956,53.8100014 L47.7974434,0 L57.7974434,22.6334489 L67.7974434,0 L91.5719913,53.8100014 L72.573561,53.8100014 Z M7.10754499,21.7630003 L16.7229599, 8.8817842e-16 L19.5285978,0 L8.51036391,24.9380633 L7.10754499, 21.7630003 Z M9.82164089,28.3034712 L22.3267926,2.0539126e-15 L26.1647839,0 L11.7406366,32.6468202 L9.82164089,28.3034712 Z M28.7229599,0 L44.5719913,0 L20.7974434,53.8100014 L16.7974434, 44.7566218 L36.5719913, 0 L28.7229599,0 L28.7229599,8.8817842e-16 Z M28.7229599,8.8817842e-16 L28.7229599, 8.8817842e-16 L34.1647839, 0 L15.6106287,41.9944523 L12.8897168,35.83609 Z M5.35664415, 14.1239332 L0,2 L10.7132883,2 L5.35664415,14.1239332 Z"
              fill="#fff"
            />
          </g>
        </svg>
      </a>
      <nav
        class="nav"
      >
        <a
          class="link active"
          href="/"
        >
          About
        </a>
        <a
          class="link"
          href="/portfolio"
        >
          Portfolio
        </a>
        <a
          class="link"
          href="/posts"
        >
          Ramblings
        </a>
        <a
          class="link"
          href="/uses"
        >
          Uses
        </a>
      </nav>
      <a
        href="https://www.github.com/visormatt"
        rel="noopener noreferrer"
        target="_blank"
        title="Find me on Github"
      >
        <svg
          class="svgOctocat"
          height="50"
          viewBox="0 0 250 250"
          width="50"
        >
          <path
            d="M0,0 L115,115 L130,115 L142,142 L250,250 L250,0 Z"
            fill="#fff"
          />
          <path
            class="octocat-arm"
            d="M128.3,109.0 C113.8,99.7 119.0,89.6 119.0,89.6 C122.0,82.7 120.5, 78.6 120.5,78.6 C119.2,72.0 123.4,76.3 123.4,76.3 C127.3,80.9 125.5, 87.3 125.5,87.3 C122.9,97.6 130.6,101.9 134.4,103.2"
            fill="#222"
          />
          <path
            class="octocat-body"
            d="M115.0,115.0 C114.9,115.1 118.7,116.5 119.8,115.4 L133.7,101.6 C136.9,99.2 139.9,98.4 142.2,98.6 C133.8,88.0 127.5,74.4 143.8,58.0 C148.5,53.4 154.0,51.2 159.7,51.0 C160.3,49.4 163.2,43.6 171.4,40.1 C171.4,40.1 176.1,42.5 178.8,56.2 C183.1,58.6 187.2,61.8 190.9,65.4 C194.5,69.0 197.7,73.2 200.1,77.6 C213.8,80.2 216.3,84.9 216.3,84.9 C212.7,93.1 206.9,96.0 205.4,96.6 C205.1,102.4 203.0,107.8 198.3, 112.5 C181.9,128.9 168.3,122.5 157.7,114.1 C157.9,116.9 156.7,120.9 152.7,124.9 L141.0,136.5 C139.8,137.7 141.6,141.9 141.8,141.8 Z"
            fill="#222"
          />
        </svg>
      </a>
    </div>
  </div>,
  <div>
    mock component
  </div>,
  <div
    class="component"
  >
    <div
      class="u-flex container copy"
    >
      Made with 
      <span
        class="spot-primary u-mx-1xs"
      >
        â™¥
      </span>
       in San Diego, CA.
    </div>
    <div
      class="component social"
    >
      <a
        class="icon"
        data-ga-label="linkedin"
        href="https://www.linkedin.com/in/matthewscholta"
        rel="noopener noreferrer"
        target="_blank"
        title="Connect on LinkedIn"
      >
        <svg
          height="50"
          viewBox="0 0 50 50"
          width="50"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            stroke="none"
            stroke-width="1"
          >
            <g
              fill-rule="nonzero"
            >
              <path
                d="M46.1478327,0 L3.67820462,0 C1.64875462,0 0,1.6086089 0, 3.5920292 L0, 46.2402381 C0,48.2253888 1.64875462,49.83642 3.67820462,49.83642 L46.1478327, 49.83642 C48.1807435,49.83642 49.8371123,48.2253888 49.8371123,46.2402381 L49.8371123,3.5920292 C49.8371123,1.6086089 48.1807435,0 46.1478327,0"
                fill="#fff"
              />
              <path
                d="M10.8247396,6.92307692 C13.1885088,6.92307692 15.1085954, 8.84350954 15.1085954,11.2076247 C15.1085954,13.5734703 13.1885088, 15.4945951 10.8247396, 15.4945951 C8.45612538, 15.4945951 6.53846154,13.5734703 6.53846154,11.2076247 C6.53846154,8.84350954 8.45612538,6.92307692 10.8247396, 6.92307692 L10.8247396, 6.92307692 Z M7.12750038,18.7457293 L14.5192104,18.7457293 L14.5192104, 42.5291228 L7.12750038, 42.5291228 L7.12750038, 18.7457293 Z"
                fill="#222"
              />
              <path
                d="M19.2307692,18.6687305 L26.3217304,18.6687305 L26.3217304, 21.9181343 L26.4200188,21.9181343 C27.4063646,20.0485764 29.8178935,18.0769231 33.4137296,18.0769231 C40.8988831, 18.0769231 42.2808058,23.0027687 42.2808058,29.4064026 L42.2808058,42.452124 L34.8942869,42.452124 L34.8942869, 30.8862674 C34.8942869,28.1269217 34.8434119,24.5795376 31.0530754,24.5795376 C27.2063269,24.5795376 26.6172881, 27.5842585 26.6172881,30.6865756 L26.6172881, 42.452124 L19.2307692,42.452124 L19.2307692,18.6687305"
                fill="#222"
              />
            </g>
          </g>
        </svg>
      </a>
      <a
        class="icon"
        data-ga-label="github"
        href="https://github.com/visormatt"
        rel="noopener noreferrer"
        target="_blank"
        title="Find me on Github"
      >
        <svg
          class="Github"
          color="#222"
          height="49"
          viewBox="0 0 50 49"
          width="50"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            stroke="none"
            stroke-width="1"
          >
            <g
              fill="#fff"
              fill-rule="nonzero"
            >
              <path
                d="M25.0023606,0 C11.195748,0 0,11.1936779 0,25.0023606 C0,36.0490583 7.16393731,45.4210103 17.0981401,48.7270289 C18.3476785,48.9584707 18.8064222, 48.1846513 18.8064222,47.5242756 C18.8064222,46.9280743 18.7832366,44.9585402 18.7724718,42.8693519 C11.816791,44.3817983 10.3490595,39.9193978 10.3490595, 39.9193978 C9.21172281,37.0294781 7.5729976,36.2610409 7.5729976,36.2610409 C5.30453473,34.7092617 7.74399135,34.7411418 7.74399135,34.7411418 C10.254661, 34.9175179 11.5766544,37.317642 11.5766544,37.317642 C13.8066125,41.1399543 17.4256367,40.0349118 18.8523794,39.3960656 C19.0767829,37.7801119 19.7247378, 36.6771396 20.439765,36.0527844 C14.8863992,35.4205626 9.0485956,33.2767225 9.0485956,23.6965141 C9.0485956,20.9668233 10.0252891,18.7364514 11.6246817, 16.9855243 C11.3650857,16.3557867 10.5092886,13.8128229 11.8668884,10.3689329 C11.8668884,10.3689329 13.9664276,9.69696439 18.7443179,12.93177 C20.7386939, 12.3777997 22.8775656,12.0999865 25.0023606,12.0904639 C27.1271555,12.0999865 29.2676834,12.3777997 31.2657854,12.93177 C36.0378794,9.69696439 38.1345202, 10.3689329 38.1345202,10.3689329 C39.4954325,13.8128229 38.6392213,16.3557867 38.3796253,16.9855243 C39.9827442,18.7364514 40.9528131,20.9668233 40.9528131, 23.6965141 C40.9528131,33.2994942 35.1038308,35.4139384 29.536388,36.0329111 C30.433174,36.8088009 31.2322491,38.3303558 31.2322491,40.6629933 C31.2322491, 44.0083447 31.2032673,46.7007727 31.2032673,47.5242756 C31.2032673,48.1896197 31.6533162,48.9692355 32.9206578,48.7237165 C42.8494783,45.4139716 50.004307, 36.045332 50.004307,25.0023606 C50.004307,11.1936779 38.8102153,0 25.0023606, 0 Z M9.36419706,35.6163904 C9.30913122,35.740599 9.1137097,35.7778615 8.93567741,35.6925718 C8.75433285,35.6110079 8.65248187,35.4416703 8.71127392, 35.3170478 C8.76509762,35.1891128 8.96093316,35.1535064 9.14186365,35.2392103 C9.32362225,35.3207739 9.42712939,35.4917678 9.36419706,35.6163904 Z M10.5940354, 36.7137588 C10.4747952,36.8243044 10.2416971,36.7729648 10.0835381,36.5982449 C9.91999679,36.4239389 9.8893587,36.1908407 10.010255,36.0786388 C10.1332215, 35.9680932 10.3592812,36.0198468 10.5232365,36.1941529 C10.6867778,36.370529 10.718658,36.601971 10.5940354,36.7137588 Z M11.437751,38.1178041 C11.2845605, 38.2242096 11.0340732,38.1244285 10.8792265,37.9020953 C10.7260358,37.6797619 10.7260358,37.4131274 10.8825387,37.306308 C11.0377994,37.1994887 11.2845605, 37.2955434 11.4414773,37.5162207 C11.5942539,37.7422802 11.5942539,38.0089145 11.437751,38.1178041 Z M12.8646548,39.7438869 C12.7276113,39.8950078 12.4357212, 39.8544328 12.2220824,39.6482467 C12.0034753,39.4466145 11.9426131,39.1605207 12.0800706,39.0094002 C12.2187701,38.8578659 12.5123164,38.9005108 12.7276113, 39.1050409 C12.9445623,39.306259 13.0108069,39.5944229 12.8646548,39.7438869 Z M14.7087894,40.2928342 C14.6483413,40.4886698 14.3672158,40.5776859 14.0840202,40.4944662 C13.8012387,40.4087624 13.6161679,40.1793907 13.6733038, 39.9814849 C13.7320959,39.7844074 14.0144634,39.6916649 14.2997291,39.7806809 C14.5820966,39.8659709 14.7675815,40.0936866 14.7087894,40.2928342 Z M16.807492, 40.5256834 C16.8145304,40.7318695 16.5743938,40.9028633 16.2771213,40.9065898 C15.9781927,40.9132142 15.7364,40.7463608 15.7330877,40.5434867 C15.7330877, 40.3352303 15.9678419,40.1658926 16.2667706,40.1609242 C16.5640432,40.1551278 16.807492,40.3207392 16.807492,40.5256834 Z M18.8692321,40.4466524 C18.9048386, 40.6478705 18.6982383,40.8544705 18.4030359,40.9095364 C18.1128019,40.9625319 17.8440972,40.8383234 17.8072487,40.6387617 C17.7712283,40.4325754 17.9815548, 40.2259751 18.2713748,40.1725657 C18.5669912,40.1212261 18.8315555,40.2421222 18.8692321,40.4466524 Z"
              />
            </g>
          </g>
        </svg>
      </a>
      <a
        class="icon"
        data-ga-label="twitter"
        href="https://twitter.com/visormatt"
        rel="noopener noreferrer"
        target="_blank"
        title="Occasionally on Twitter"
      >
        <svg
          color="#222"
          height="41"
          viewBox="0 0 50 41"
          width="50"
        >
          <g
            fill="none"
            fill-rule="evenodd"
            stroke="none"
            stroke-width="1"
          >
            <g
              fill="#fff"
              fill-rule="nonzero"
            >
              <path
                d="M49.9996169,4.97067009 C48.1599515,5.78653128 46.1829373, 6.33793714 44.1078956,6.5859326 C46.2259159,5.31632748 47.8523345, 3.30602797 48.6182675, 0.910750218 C46.6363155,2.08616825 44.4409339,2.93988716 42.1043629,3.40003215 C40.2330579,1.40637538 37.5671067,0.160545645 34.6167654,0.160545645 C28.9514588, 0.160545645 24.3584219,4.75339973 24.3584219,10.4183404 C24.3584219, 11.2223139 24.4493169,12.0052553 24.6241574,12.7560086 C16.0986732, 12.3282347 8.54011526, 8.24435655 3.48071512,2.03825172 C2.59773421,3.55329187 2.09186737,5.31541306 2.09186737,7.19531379 C2.09186737,10.7541219 3.9030024,13.8939347 6.65545945, 15.7334173 C4.97380878,15.680197 3.39219756,15.2187718 2.00901931,14.4504614 C2.00828777,14.4932571 2.00828777,14.5362356 2.00828777,14.5793971 C2.00828777, 19.549548 5.54423492,23.6956081 10.2369456,24.6378444 C9.37609407,24.8723063 8.46988644,24.9975842 7.53423394,24.9975842 C6.87327846,24.9975842 6.23061175, 24.9333906 5.6044049,24.8135993 C6.90967309,28.8890648 10.698005,31.8547686 15.1867961,31.9376166 C11.6760874,34.6887934 7.25295294,36.3287458 2.4470349, 36.3287458 C1.61910314,36.3287458 0.802510401,36.2802806 0,36.1855448 C4.53963383,39.0960166 9.93170633,40.7941272 15.7246683,40.7941272 C34.592807, 40.7941272 44.9105889,25.1632802 44.9105889,11.6078408 C44.9105889,11.1630584 44.900713,10.7206535 44.8809612,10.280809 C46.8850425,8.83453455 48.6243028, 7.02778882 49.9996169,4.97067009"
              />
            </g>
          </g>
        </svg>
      </a>
    </div>
  </div>,
  <div
    class="Toastify"
  />,
]
`;
